File: medium/1493.java
class Solution {
    public int longestSubarray(int[] nums) {
        int prevLen = 0;
        boolean isZero = false;
        int currLen = 0;
        int ans = 0;
        for (int i : nums) {
            if (i == 0) {
                isZero = true;
                prevLen = currLen;
                currLen = 0;
            } else
                currLen++;

            ans = Math.max(ans, prevLen + currLen);

        }

        if (isZero)
            return ans;

        return ans - 1;
    }
}

File: medium/1493.py
class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        result =0
        currLen=0
        prevLen=0
        isZero=False
        for i in nums:
            if i==0:
                isZero=True
                prevLen = currLen
                currLen=0
            else:
                currLen+=1
             
            result = max(result, currLen+prevLen)
            
        if isZero:
            return result
        return result-1;        